raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -155,7 +156,12 @@ def handle_cancelroute(
         if sub_iteration.new_state:
             payment_state.initiator = sub_iteration.new_state
         else:
-            payment_state = None
+            # Here we don't delete the initiator state, but instead let it live.
+            # It will be deleted when the lock expires. We do that so that we
+            # still have an initiator payment task around to process the
+            # LockExpired message that our partner will send us.
+            # https://github.com/raiden-network/raiden/issues/3146#issuecomment-447378046
+            payment_state.initiator = old_initiator_route
```

Comment:
IMO, I think it would be more structured to pass in the `initiator_state` into `try_new_route` and instead of returning None in:
https://github.com/raiden-network/raiden/blob/53b8eee062e3e309486ea438c9487f349b964c11/raiden/transfer/mediated_transfer/initiator.py#L178-L193
We would return `initiator_state` that was originally passed. This follows the convention of the code base rather than saving the state in a temp variable and then reusing it like you did.

Reply:
Sure. I was also thinking of doing it like that but opted for a temp variable as the quick solution to test the hypothesis.